packages/core/src/context/starknet.tsx

```diff
@@ -31,6 +33,8 @@ export interface StarknetState {
   disconnect: () => Promise<void>;
   /** List of registered connectors. */
   connectors: Connector[];
+  /** Current explorer factory. */
+  explorer?: ExplorerFactory<StarkcompassExplorer | StarkscanExplorer | ViewblockExplorer | VoyagerExplorer>;
```

Comment:
We should keep the explorer type generic so that users can provide their own implementation.

```suggestion
  explorer?: ExplorerFactory<T>;
```